onnxruntime/core/platform/android/cxa_demangle.cc

```diff
@@ -0,0 +1,64 @@
+// #if USE_DUMMY_EXA_DEMANGLE
+
+extern "C" {
+#include <stdlib.h>
+#include <string.h>
+#include <limits.h>
+
+#define DEMANGLE_SUCCESS 0
+#define DEMANGLE_MEMORY_ALLOCATION_FAILURE -1
+#define DEMANGLE_INVALID_MANGLED_NAME -2
+#define DEMANGLE_INVALID_ARGUMENTS -3
+
+/**
+ * An dummy implementation of __cxa_demangle https://itanium-cxx-abi.github.io/cxx-abi/abi.html#demangler It is
+ * referred in https://github.com/llvm/llvm-project/blob/d09d297c5d/libcxxabi/src/cxa_default_handlers.cpp#L53 However
+ * it contribute a large percentage of binary size in minimal build. To avoid it, the LLVM user can compile libc++abi.a
+ * with LIBCXXABI_NON_DEMANGLING_TERMINATE defined, see https://reviews.llvm.org/D88189 But this make the compilation
+ * process extremely complex. Here we provide a dummy __cxa_demangle. It always copy the mangled name to output buffer.
+ */
+char* __cxa_demangle(const char* mangled_name, char* buf, size_t* n, int* status) {
+  if (!status) return NULL;
+  if (!mangled_name ||                             // Input must be valid
+      (buf && !n) ||                               // If it is non-null, then n must also be nonnull
+      (buf && n && ((*n == 0) || (*n > INT_MAX)))  // if buf and n is supplied, *n must be a 'valid' value
+  ) {
+    *status = DEMANGLE_INVALID_ARGUMENTS;
+    return NULL;
+  }
+
+  // size is the buffer size, including the null char
+  size_t output_size = n ? *n : 0;
+  // It is caller's responsibility to ensure mangled_name is null-terminated as it is required in the API doc.
+  size_t input_size = strlen(mangled_name) + 1;
+  if (input_size > output_size) {
+    free(buf);
+    buf = NULL;
```

Comment:
See the last line of the ABI doc, it says:

If buf is not big enough to store the resulting demangled name, __cxa_demangle must _either_ **a) call free to deallocate buf and then allocate a new buffer with malloc**, or b) call realloc to increase the size of the buffer. In either case, the new buffer size will be stored in *n.

wrt the cxa_default_handlers.cpp, I think it is the implementor is sure about the symbol length is always less than 1024?

Reply:
Maybe they are sure it's always less than 1024. However, given that usage of passing in a pointer to a stack allocated buffer I don't think it's safe for the implementation to assume it can call `free` (despite what the documentation says) as that will crash at runtime. The documentation isn't going to stop it crashing :P.